app/src/frontend/building/data-components/data-entry.tsx

```diff
@@ -20,7 +37,13 @@ const DataEntry: React.FunctionComponent<any> = (props) => { // TODO: remove any
                 maxLength={props.maxLength}
                 disabled={props.mode === 'view' || props.disabled}
                 placeholder={props.placeholder}
-                onChange={props.onChange}
+                onChange={e => {
+                    const transform = props.valueTransform || (x => x);
+                    const val = e.target.value === '' ?
+                                    null :
+                                    transform(e.target.value);
```

Comment:
DataEntry allows supplying a valueTransform function, now used only with the postcode field - allows to add custom value processing logic like `value => value.toUpperCase()` for certain fields.